chalice/app.py

```diff
@@ -100,6 +100,54 @@ def __repr__(self):
         return 'CaseInsensitiveMapping(%s)' % repr(self._dict)
 
 
+class CORSConfig(object):
+    """A cors configuration to attach to a route."""
+
+    _REQUIRED_HEADERS = ['Content-Type', 'X-Amz-Date', 'Authorization',
+                         'X-Api-Key', 'X-Amz-Security-Token']
+
+    def __init__(self, allow_origin='*', allow_headers=None,
+                 expose_headers=None, max_age=None, allow_credentials=None):
+        self.allow_origin = allow_origin
+
+        if allow_headers is None:
+            allow_headers = list(self._REQUIRED_HEADERS)
+        else:
+            allow_headers.extend(self._REQUIRED_HEADERS)
```

Comment:
What if a passed in `allow_header` contains one of the `_REQUIRED_HEADERS`?  Seems like we should check for existing values or use a `set()`.